### ISSUE TITLE
Allow multiple logo images

### ISSUE THREAD

**Issue by mikeapp** [IIIF/trc#3]
At issue in IIIF/api#1760 is whether the `logo` property value should be an array.  Use cases that would be supported by allowing multiple logo images include:
* providing alternate versions of logos that contain internationalized text
* providing static logo images at varying sizes

In the Alpha draft spec the `logo` value _MUST_ be an array, as suggested here.  More recently the Editors updated the Beta draft to allow a maximum of one logo (IIIF/api#1698) - that is, that he value _MUST_ be a JSON object.  The proposal is to reverse this decision.

Alpha draft:
https://iiif.io/api/presentation/3.0/#logo

(Note that the `logo` examples in section 5.2 and Appendix B of the Alpha draft are inconsistent with the requirement to use an array.)

**Comment by ahankinson** [comment]
What is the mechanism for clients being able to choose the logo to display (use case B)? Since clients MUST render the logo, it doesn’t seem correct that they must render every logo.

**Comment by mikeapp** [comment]
Correct, this entails an adjustment to the client requirements - I'll edit the parent to clarify.  The intent is that one image from the array would be rendered.  The client would apply its own selection criteria.

**Comment by azaroth42** [comment]
Summary on the call: We propose to allow multiple logo images. If you look at the alpha draft you might think that's what we have, but in beta we introduced provider, each of which has only one logo.  Link to relevant piece: https://preview.iiif.io/api/image-prezi-rc2/api/presentation/3.0/#provider 

We thought there should be one logo per provider, the original multiple logos was to allow multiple providers. A question came up that if for any provider they might have multiple logos, eg internationalized text in different images, or a light and dark version, a logo that's fixed, and one with an image service.  Clients could look at these and other properties of the image to make a decision about the best one to render in the current context. If we restrict to one, it would be a breaking change to add this in later.  So we want to allow multiple logos for this sort of customization.  The simple use case of a single logo is still possible, of course, and covers most people.  This gives us extension possibility, avoids a breaking change.

**Comment by azaroth42** [comment]
Question: How to make a decision about which logo to use?
Answer: Implementation dependent. Could recommend selecting the first one, but that's not really part of this issue.

**Comment by glenrobson** [comment]
I'm -1 as without defining properties to programmatically work out which logo to use it doesn't seem that useful. If we were to give examples with language or an extension that defines the light or dark property I would be plus 1.

**Comment by irv** [comment]
i have this exact use-case (light and dark version of a logo). i'm happy for the rules to be another issue, though

**Comment by Siani81** [comment]
We have this use case also, different logo for mobile browsers, +1 for a new issue for rules

**Comment by ahankinson** [comment]
As an implementer, I’m getting quite mixed up over when I should use a provider block and when I should use a standalone logo block. 

I’m also -1 on this without guidance on how viewers are to select the appropriate logo. It seems like it’s opening up the space for diverging application, and since the logo is a MUST requirement, I think it’s a part of the spec where we should be conservative in what we require.

**Comment by mikeapp** [comment]
The `logo` would always be within the `provider` (at least in Prezi, see IIIF/api#1735 for Image).  Each provider could have 0..n logo images.

**Comment by dismorfo** [comment]
I'm also mixed up over this one. I agree with @ahankinson and @glenrobson

**Comment by azaroth42** [comment]
To me, the issue is whether we should have multiple logos per provider or not. The details of how to choose which logo can then be discussed in a separate issue and then brought to the TRC for approval when it's ready. But until multiple is approved or not, we can't move forwards to that discussion.

**Comment by scossu** [comment]
I think we have a chiken-and-egg situation. The utility of the feature seems clear, but at the same time a clear method (not necessarily the only one) to select from an unordered list of logos should be defined; otherwise the risk would be that the introduction of this feature may lead to confusion in its implementation.

**Comment by azaroth42** [comment]
@scossu I think this is the egg, and in order for the chicken to survive, we need the additional information. But without the ability to have multiple logos (this issue), there's no need to have a method to select which one to display (the raised issue).

**Comment by regisrob** [comment]
I agree with the egg. I'm just wondering about the need to clarify the usage of "multiple logos" in the spec, i.e. between "multiple logos representing the different providers of my Manifest" (i.e. owning/hosting/funding institutions) and "multiple providers, each of which having potentially multiple variants of its own logo".

If I have a logo for the digital library and another one for the library that holds the physical object, what am I supposed to do?:
* only put my preferred one in manifest.logo (given that most clients will only pick this one in many contexts)?
* put the two logos in manifest.logo and let the client choose one (or give it a hint to pick my preferred one)?
* in any case put the logos of each institution in a `provider` array (which will most probably by displayed in some kind of metadata panel by the IIIF viewers)?

(this may rather be detailed in a cookbook recipe...)

**Comment by rsinghal** [comment]
I am +1 so long as it is tied to parameters around choosing the logo, so not entirely sure how to vote.

**Comment by mixterj** [comment]
I am not opposed to the idea of multiple logos but I am not comfortable with the ramifications of not clearly articulating how a data consumer should interoperate/use an array of multiple logos. The added use case of multiple versions of a log (color and grayscale) is even more problematic - if the data producer does not care which is used. This is not the same use case as multiple logos that are to be used in specific instances or the other use case of multiple logos representing multiple organizations contributing images to the same manifest (assuming they are not connected the the individual images)

Until these use cases are sorted out and their ramifications for data producers and consumers are defined I am a -1

**Comment by azaroth42** [comment]
### Issue 3 (Allow multiple logo images)
  +1: 26 [Siani81 aisaac azaroth42 beaudet cubap emulatingkat gigamorph irv jbhoward-dublin jonhartzler joshuago78 jronallo jtweed julsraemy jwd markpatton mattmcgrattan mcwhitaker mikeapp regisrob rentonsa scossu sredick tomcrane tpendragon zimeon]
   0: 1 [hadro]
  -1: 9 [ahankinson andrewgunther awead dismorfo glenrobson ipf kzhr mejackreed mixterj]

### Result
Super majority is in favor, issue is approved
